core/carp_string.h

```diff
@@ -22,20 +22,13 @@ void String_delete(String s) {
 }
 
 void String_string_MINUS_set_BANG_(String *s, int i, char ch) {
-    int l = strlen(*s);
-    (void)l;
-    assert(i >= 0);
-    assert(i < l);
+    assert((size_t)i < strlen(*s));
```

Comment:
Should we not check for non-negativity here as well?

Reply:
`size_t` is unsigned, so it shouldn't be needed. There might be problems with that assert for near-4GB strings, but the solution to that would be to use size_t for string sizes/offsets (that would be another PR).

Reply:
I guess, but since we’re passing in an `int` from Carp, you might end up with errors that aren’t super useful, right? I mean I guess if it fails it fails, but if I pass in `-1` I’d still be happier if my assert failed on `i >= 0` than `i < strlen(*s)`.

Reply:
I guess both assertions are useless because you don't know what the value is. `i` could be negative just because it's an overflow of a previous operation. Let me think about this, maybe it would be better to have something built on top of `assert` to display values.